src/tests/build.proj

```diff
@@ -467,6 +469,88 @@
     <Message Importance="High" Text="App: $(FinalPath)" />
   </Target>
 
+
+  <Target Name="_SetCustomIlcProps">
+    <PropertyGroup>
+      <_IsApplePlatform>true</_IsApplePlatform>
+      <_IsiOSLikePlatform>true</_IsiOSLikePlatform>
+      <_targetOS>$(TargetOS)</_targetOS>
+      <UseNativeAOTRuntime>true</UseNativeAOTRuntime>
+      <NativeLib>static</NativeLib>
+      <CustomNativeMain>true</CustomNativeMain>
+      <_SymbolPrefix>_</_SymbolPrefix>
+      <DsymUtilOptions>--flat</DsymUtilOptions>
+      <IlcPath>$([MSBuild]::NormalizeDirectory('$(RepoRoot)', 'artifacts', 'bin', 'coreclr', '$(HostOS).$(BuildArchitecture).$(CoreCLRConfiguration)', 'ilc'))</IlcPath>
+      <IlcToolsPath>$(IlcPath)</IlcToolsPath>
+      <IlcSdkPath>$([MSBuild]::NormalizeDirectory('$(RepoRoot)', 'artifacts', 'bin', 'coreclr', '$(TargetOS).$(TargetArchitecture).$(CoreCLRConfiguration)', 'aotsdk'))</IlcSdkPath>
+      <IlcFrameworkPath>$(LibrariesAllBinArtifactsPath)</IlcFrameworkPath>
+      <IlcFrameworkNativePath>$(LibrariesAllBinArtifactsPath)</IlcFrameworkNativePath>
+    </PropertyGroup>
+  </Target>
+
+  <Target Name="BuildNativeAOTiOSApp" DependsOnTargets="_SetCustomIlcProps;SetupOSSpecificProps">
+    <PropertyGroup>
+      <CMDDIR_Grandparent>$([System.IO.Path]::GetDirectoryName($([System.IO.Path]::GetDirectoryName($(_CMDDIR)))))</CMDDIR_Grandparent>
+      <TestingGroup>$([System.IO.Path]::GetFileName($(CMDDIR_Grandparent)))</TestingGroup>
+      <CategoryWithSlash>$(TestingGroup)/$([System.String]::Copy('$(_CMDDIR)').Replace("$(CMDDIR_Grandparent)/",""))</CategoryWithSlash>
+      <TestName>$([System.IO.Path]::GetFileName($(_CMDDIR)))</TestName>
+      <Category>$([System.String]::Copy('$(CategoryWithSlash)').Replace('/','_'))</Category>
+      <BuildDir>$(IntermediateOutputPath)\iOSApps\$(Category)</BuildDir>
+      <FinalPath>$(XUnitTestBinBase)$(CategoryWithSlash)\$(Category).app</FinalPath>
+    </PropertyGroup>
+
+    <PropertyGroup>
+      <BundleDir>$([MSBuild]::NormalizeDirectory('$(BuildDir)', 'AppBundle'))</BundleDir>
+      <AssemblyName>$(Category)</AssemblyName>
+      <_NativeDependencies>$(IntermediateOutputPath)\..\$(CategoryWithSlash)\$(TestName)\native\$(TestName).o</_NativeDependencies>
+      <DevTeamProvisioning>-</DevTeamProvisioning>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <_LinkerFlagsToDrop Include="@(NativeFramework->'-framework %(Identity)')" />
+      <_LinkerFlagsToDrop Include="@(LinkerArg)" Condition="$([System.String]::new('%(Identity)').Contains('swift'))" />
+      <LinkerArg Remove="@(_LinkerFlagsToDrop)" />
+      <ExtraLinkerArguments Include="@(LinkerArg)" />
+    </ItemGroup>
+
+    <RemoveDir Directories="$(BundleDir)" />
+    <RemoveDir Directories="$(BuildDir)" />
+    <RemoveDir Directories="$(AppDir)" />
+    <MakeDir Directories="$(BuildDir)" />
+
+    <AppleAppBuilderTask
```

Comment:
Couldn't we use `AppleBuild.targets` ?

Reply:
Good idea. There is the `_AppleGenerateAppBundle` target that may be updated to support both mono and coreclr runtime flavors.